### PR TITLE
Fix/no anonymous functions

### LINKED ISSUE #4369: Make anonymous function components named
In all our projects, code like:
```tsx
export const AboutModalBoxCloseButton = ({
  className = '',
  onClose = () => undefined as any,
  'aria-label': ariaLabel = 'Close Dialog',
  ...props
}: AboutModalBoxCloseButtonProps) => ( ... );
```
gets compiled to anonymous CJS arrow functions:
```js
exports.AboutModalBoxCloseButton = (_a) => { ... };
```

However, for Enzyme (which uses CJS), we'd like our code to get compiled down to named components so we can write selectors like `view.find(AboutModalBoxCloseButton)`.

### ~~Named functions~~
This has been ruled out in favor of display names.

~~An alternative way to write our arrow functions to accomplish this is like:~~
```tsx
export function AboutModalBoxCloseButton({
  className = '',
  onClose = () => undefined as any,
  'aria-label': ariaLabel = 'Close Dialog',
  ...props
}: AboutModalBoxCloseButtonProps) {
  return ( ...  );
}
```
~~which gets compiled to CJS as a named function:~~
```js
function AboutModalBoxCloseButton(_a) { ... }
```

~~The types change slightly from:~~
```ts
export declare const AboutModalBoxCloseButton: React.FunctionComponent<AboutModalBoxCloseButtonProps>;
```
~~to~~
```ts
export declare function AboutModalBoxCloseButton({ className, onClose, 'aria-label': ariaLabel, ...props }: AboutModalBoxCloseButtonProps): JSX.Element;
```

~~... but these are compatible and _should_ be okay (see [blog post](https://blog.echobind.com/react-with-typescript-components-as-function-declarations-vs-function-expressions-e433ac8d6938) and [React.FC typing README](https://github.com/typescript-cheatsheets/react-typescript-cheatsheet/blob/master/README.md#function-components))~~

### Display names

An alternative way to write our arrow functions to accomplish this is like:
```tsx
export const AboutModalBoxCloseButton = ({
  className = '',
  onClose = () => undefined as any,
  'aria-label': ariaLabel = 'Close Dialog',
  ...props
}: AboutModalBoxCloseButtonProps) => ( ... );
AboutModalBoxCloseButton.displayName = 'AboutModalBoxCloseButton';
```

This works everywhere and the function name won't get obfuscated by Webpack's terser or other build tools so that React dev tools can pickup component names in our docs.

### PR DESCRIPTION
**Pull request by redallen**
**What**: Add an eslint rule to disallow anonymous functions for Enzyme and React dev tools' sake. Closes #4369

**Additional issues**:

### CODE REVIEW

File: packages/react-core/src/components/DataList/DataListToggle.tsx [hunk 1]
```diff
@@ -21,29 +21,34 @@ export interface DataListToggleProps extends React.HTMLProps<HTMLDivElement> {
   'aria-controls'?: string;
 }
 
-export const DataListToggle: React.FunctionComponent<DataListToggleProps> = ({
+/**
```
Comment (seanforyou23) [inline packages/react-core/src/components/DataList/DataListToggle.tsx:24]: What's the value in adding so many instances of this empty comment block? If it's only to satisfy eslint, we should handle it another way. We could either actually add text to these comments so they serve a purpose, or we could exclude this from being something that eslint requires. I think either way would be better than bloating the codebase with the empty comments.

File: test.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const fs = require('fs');
+const reactDocgen = require('react-docgen');
+
+const sourcePath = require.resolve('@patternfly/react-catalog-view-extension/src/components/CatalogTile/CatalogTileBadge.tsx');
+const sourceText = fs.readFileSync(sourcePath, 'utf8');
+
+
```
Comment (seanforyou23) [inline test.js:7]: Can remove space

File: test.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+const parsedComponents = reactDocgen.parse(
+  sourceText,
+  reactDocgen.resolver.findAllExportedComponentDefinitions,
+  null,
+  { filename: sourcePath }
+);
+
+console.dir(parsedComponents, { depth: 4 })
```
Comment (seanforyou23) [inline test.js:15]: Can we add a newline?

File: packages/eslint-plugin-patternfly-react/lib/rules/no-anonymous-functions.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            fix(fixer) {
+              const declarationNode = context.getSourceCode().getFirstToken(node.declaration);
+              const initBody = declaration.init.body;
+              const equalsToken = context.getSourceCode().getTokenAfter(declaration.id);
+              const fixes = [
+                fixer.replaceText(declarationNode, 'function'), // export (const|var) -> export function
+                fixer.remove(typeAnnotation), // remove : React.FunctionComponent<AboutModalBoxCloseButtonProps>
+                fixer.remove(equalsToken) // remove =
+              ];
+              // () => (<div></div>) -> function() { return (<div></div>) }
```
Comment (seanforyou23) [inline packages/eslint-plugin-patternfly-react/lib/rules/no-anonymous-functions.js:50]: Can we remove it?

**Comment by redallen** [comment]
Closing this in favor of adding `displayName`s. That should address @jschuler 's feedback in Slack that the production builds are missing component names in the React devtools and @seanforyou23 's feedback about the empty comment blocks

**Pull request closed without merging**